src/py/flwr/client/grpc_rere_client/connection.py

```diff
@@ -172,17 +168,22 @@ def send(client_message_proto: ClientMessage) -> None:
             return
         task_ins: TaskIns = cast(TaskIns, state[KEY_TASK_INS])
 
-        # Wrap ClientMessage in TaskRes
-        task_res = TaskRes(
-            task_id="",  # This will be generated by the server
-            group_id=task_ins.group_id,
-            workload_id=task_ins.workload_id,
-            task=Task(
+        # fill `group_id` and `workload_id` in TaskRes
+        task_res.MergeFrom(
+            TaskRes(
+                task_id="",  # This will be generated by the server
+                group_id=task_ins.group_id,
+                workload_id=task_ins.workload_id,
+            )
+        )
+
+        # fill `producer`, `consumer`, and `ancestry` in Task
+        task_res.task.MergeFrom(
```

Comment:
The behaviour of `MergeFrom` is non-intuitive. I think we should add a comment here. I am also not sure, but maybe add a link to the docs.

Afaik fields will be overwritten, lists will be concatenated, and sub-messages will be recursively merged. I can imagine this to result in subtle and hard-to-debug bugs.

Reply:
Hi @tanertopal ! Thanks so much for pointing that out. I've checked the API documentation and it is true that certain fields may not be overwritten when calling `MergeFrom`. 
I've pushed a new commit, adding `ClearField` before `MergeFrom` and some comments.